### PR TITLE
add benchmark: Lazy vs. Uthread

### PR DESCRIPTION
**Pull request by PikachuHyA**
## Why

add benchmark to compare Lazy and Thread

## What is changing

- add call depth benchmark
- add read file benchmark
- add switch benchmark

### CODE REVIEW

File: benchmarks/CallDepth.bench.cpp [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        state.PauseTiming();
+        std::vector<std::function<void()>> task_list;
+        task_list.reserve(task_num);
+        std::atomic<bool> running = true;
+        int m = 1;
+        for (int i = 0; i < task_num; ++i) {
+            task_list.emplace_back([n, m, s, e = &e]() {
+                Service service{
+                    .n = n, .m = m, .filename_ = s.c_str(), .e_ = e};
+                service.run();
```
Comment (RainMark) [inline benchmarks/CallDepth.bench.cpp:164]: coroutine call depth perf test should not include the io path?

File: benchmarks/ReadFileUtil.hpp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    task_list.reserve(num);
+    for (int i = 0; i < num; ++i) {
+        task_list.template emplace_back(
+            [&s, e]() { Uthread_read_file(s.c_str(), e); });
+    }
+    collectAll<Launch::Schedule>(task_list.begin(), task_list.end(), e);
+}
+
+template <typename FileDescriptor, typename Buffer, typename Executor>
+struct ReadSomeAwaiter {
```
Comment (RainMark) [inline benchmarks/ReadFileUtil.hpp:65]: try to use FutureAwaiter to instead of self-defined awaiter?

File: benchmarks/CallDepth.bench.cpp [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        auto f = [&s, n, e = &e]() mutable -> Lazy<void> {
+            std::vector<Lazy<uint64_t>> lazy_list;
+            lazy_list.reserve(task_num);
+            int m = 1;
+            for (int i = 0; i < task_num; ++i) {
+                lazy_list.push_back(
+                    AsyncService().run(s.c_str(), n, e->getIOExecutor(), m));
+            }
+            [[maybe_unused]] auto ret =
+                co_await collectAllPara(std::move(lazy_list));
```
Comment (RainMark) [inline benchmarks/CallDepth.bench.cpp:198]: also should not include task schedule latency, collectAllPara will occurred the task schedule.

File: benchmarks/CallDepth.bench.cpp [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+using namespace async_simple;
+using namespace async_simple::coro;
+using namespace async_simple::executors;
+using namespace async_simple::uthread;
+namespace fs = std::filesystem;
+static int task_num = 10;
+struct Service {
+    void run() { service1(); }
+
+    void service1() {
```
Comment (RainMark) [inline benchmarks/CallDepth.bench.cpp:34]: we already has call depth test case in https://github.com/alibaba/async_simple/blob/main/async_simple/coro/test/LazyTest.cpp#L1307, maybe you can move to benchmark module. And add the other perf test like coroutine switch.

File: benchmarks/ReadFileUtil.hpp (hunk 2 above)
Comment (PikachuHyA) [reply benchmarks/ReadFileUtil.hpp]: done

**Review by ChuanqiXu9 — APPROVED** [review]
LGTM.

**Pull request merged**